vms/platformvm/state/camino_deposit.go

```diff
@@ -1,72 +1,236 @@
 
 package state
 
 import (
+	"encoding/binary"
 	"fmt"
+	"time"
 
 	"github.com/ava-labs/avalanchego/database"
 	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils"
 	"github.com/ava-labs/avalanchego/vms/platformvm/blocks"
 	"github.com/ava-labs/avalanchego/vms/platformvm/deposit"
 )
 
-func (cs *caminoState) UpdateDeposit(depositTxID ids.ID, deposit *deposit.Deposit) {
+func (cs *caminoState) AddDeposit(depositTxID ids.ID, deposit *deposit.Deposit) {
+	cs.addedDeposits[depositTxID] = deposit
+}
+
+func (cs *caminoState) ModifyDeposit(depositTxID ids.ID, deposit *deposit.Deposit) {
 	cs.modifiedDeposits[depositTxID] = deposit
 	cs.depositsCache.Evict(depositTxID)
 }
 
+func (cs *caminoState) RemoveDeposit(depositTxID ids.ID, deposit *deposit.Deposit) {
+	cs.removedDeposits[depositTxID] = deposit
+	cs.depositsCache.Evict(depositTxID)
+}
+
 func (cs *caminoState) GetDeposit(depositTxID ids.ID) (*deposit.Deposit, error) {
-	// Try to get from modified state
-	d, ok := cs.modifiedDeposits[depositTxID]
-	// deposit was deleted
-	if ok && d == nil {
+	if _, ok := cs.removedDeposits[depositTxID]; ok {
 		return nil, database.ErrNotFound
 	}
-	// Try to get it from cache
-	if !ok {
-		var depositIntf interface{}
-		if depositIntf, ok = cs.depositsCache.Get(depositTxID); ok {
-			d = depositIntf.(*deposit.Deposit)
-		}
+	if deposit, ok := cs.modifiedDeposits[depositTxID]; ok {
+		return deposit, nil
+	}
+	if deposit, ok := cs.addedDeposits[depositTxID]; ok {
+		return deposit, nil
 	}
-	// Try to get it from database
-	if !ok {
-		depositBytes, err := cs.depositsDB.Get(depositTxID[:])
-		if err != nil {
-			return nil, err
-		}
 
-		d = &deposit.Deposit{}
-		if _, err := blocks.GenesisCodec.Unmarshal(depositBytes, d); err != nil {
-			return nil, err
-		}
+	if depositIntf, ok := cs.depositsCache.Get(depositTxID); ok {
+		return depositIntf.(*deposit.Deposit), nil
+	}
 
-		cs.depositsCache.Put(depositTxID, d)
+	depositBytes, err := cs.depositsDB.Get(depositTxID[:])
+	if err != nil {
+		return nil, err
 	}
 
+	d := &deposit.Deposit{}
+	if _, err := blocks.GenesisCodec.Unmarshal(depositBytes, d); err != nil {
+		return nil, err
+	}
+
+	cs.depositsCache.Put(depositTxID, d)
+
 	return d, nil
 }
 
+func (cs *caminoState) GetNextToUnlockDepositTime() (time.Time, error) {
+	if cs.depositsNextToUnlockTime == nil {
+		return time.Time{}, database.ErrNotFound
+	}
+	return *cs.depositsNextToUnlockTime, nil
+}
+
+func (cs *caminoState) GetNextToUnlockDepositIDsAndTime() ([]ids.ID, time.Time, error) {
+	if cs.depositsNextToUnlockTime == nil {
+		return nil, time.Time{}, database.ErrNotFound
+	}
+	return cs.depositsNextToUnlockIDs, *cs.depositsNextToUnlockTime, nil
+}
+
 func (cs *caminoState) writeDeposits() error {
-	for depositTxID, deposit := range cs.modifiedDeposits {
-		delete(cs.modifiedDeposits, depositTxID)
+	// checking if all current deposits were removed
+	notAllNextToUnlockRemoved := false
```

Comment:
`notAllNextToUnlockRemoved := false` please revert to the
`anyNextToUnlockRemained := true`
Negating is usually harder to get think of - wdyt?